xmalloc.c

```diff
@@ -21,64 +21,55 @@
  *    Email: <neilb@suse.de>
  */
 
-#include	"mdadm.h"
-/*#include	<sys/socket.h>
-#include	<sys/utsname.h>
-#include	<sys/wait.h>
-#include	<sys/un.h>
-#include	<ctype.h>
-#include	<dirent.h>
-#include	<signal.h>
-*/
+#include "xmalloc.h"
+
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+static void *exit_memory_alloc_failure(void)
+{
+	fprintf(stderr, "Memory allocation failure - aborting\n");
+	exit(5);
```

Comment:
Why 5? Maybe we can consider exit with 1 for failure?

Reply:
I wonder about that too. If this is for legacy reasons, I'd appreciate a comment why this specific value is used.